### DISCUSSION TITLE
[question] supabase-js sdk insert/upsert optional parameters

### DISCUSSION THREAD

**Discussion by proton1k** [supabase/supabase#8096] [Questions] [answered]
**TL;DR**

1. If I send a 1st message in the channel, it breaks due to [this](https://github.com/supabase/supabase/blob/5cd1f2b480d504782b81028ae5772e2311a69f02/examples/slack-clone/nextjs-slack-clone/components/Message.js#L20) code `<p className="text-blue-700 font-bold">{message.author.username}</p>`
2. Is there a way of [conditional returning](https://github.com/supabase/supabase-js/blob/4438e1783ce87d651edfa158d710cb8bf00e8775/example/next-storage/components/Account.tsx#L112) with some filters? There's too little info in [documentation](https://supabase.com/docs/reference/javascript/insert#notes).

-------

**Main message**

Hi, I'm checking out the slack clone example [here](https://github.com/supabase/supabase/blob/b5fdbb7d02b381e28d5f8090616e61215e8d2c1e/examples/nextjs-slack-clone/lib/Store.js#L192) and stumbled upon an issue that on a new message created, the Message.js component throws error that `message.user` is undefined.
It happens because there's no user data in `users` store yet when it's my message and I send a 1st message. Here's the code I investigate:
```js
      .insert([{ message, channel_id, user_id }])
```
and seems that it's not possible to pass any other postgrest parameters to actually return something back like this:
```js
      .insert([{ message, channel_id, user_id }], {
        returning: `*, user:user_id(*)`,
      })
```
right? Or is there a way?

**Comment by steve-chavez** [comment, marked as answer]
Yes, this is possible with `select()` not `returning`. 

```js
.insert([{ message, channel_id, user_id }])
.select(`*, user:user_id(*)`)
```

See this previous discussion https://github.com/supabase/supabase/discussions/7942

**Reply by proton1k** [reply]
Apparently, I miss the required header `Prefer: return=representation` to return some results back.
Final code would look like:

```
.insert([{ message, channel_id, user_id }], { returning: "representation" })
.select(`*, user:user_id(*)`)
```
